front/src/applications/operationalStudies/types.ts

```diff
@@ -152,3 +155,26 @@ export type TimetableItemRoundTripGroups = {
 export type StudyCardDetails = SearchResultItemStudy | StudyWithScenarios;
 
 export type ScenarioCardDetails = SearchResultItemScenario | ScenarioWithDetails;
+// Unified operational point format for both projection cases
+export type ProjectionOperationalPoint = {
+  extensions?: OperationalPointExtensions;
+  id: string;
+  part: OperationalPointPart;
+  position: number;
+  weight: number | null;
+};
+
+export type ProjectionWithPathFinding = {
+  pathfinding: PathfindingResultSuccess | undefined;
+  geometry?: GeoJsonLineString | null;
+  operationalPoints?: ProjectionOperationalPoint[];
+  pathUsedForProjection?: TrainSchedule['path'] | undefined;
+  projectingOnSimulatedPathException?: boolean;
+};
+
+export type ProjectionWithoutPathFinding = {
+  operationalPoints: ProjectionOperationalPoint[];
+  pathUsedForProjection?: TrainSchedule['path'] | undefined;
+};
+
+export type PathProjectionResult = ProjectionWithPathFinding | ProjectionWithoutPathFinding;
```

Comment:
Could we change these types like this ?
```suggestion
export type PathProjectionResult = {
  pathUsedForProjection: TrainSchedule['path'];
  operationalPoints: ProjectionOperationalPoint[];
} & (
  | {
      type: 'successful_pathfinding';
      pathfinding: PathfindingResultSuccess;
      geometry: GeoJsonLineString;
      pathUsedForProjection: TrainSchedule['path'];
      projectingOnSimulatedPathException?: boolean;
    }
  | {
      type: 'failed_pathfinding';
    }
);
```

- Even if the pathfinding is not successful, we should have the path of the timetableItem and the Ops corresponding to this path inputs.
- It's easier to differentiate the 2 cases if you introduce a type (it allows to avoid test `'pathfinding' in ...` which are more fragile).